apps/web/src/contexts/language-context.tsx

```diff
@@ -0,0 +1,101 @@
+import { DEFAULT_LANG, LANG_COOKIE, RTL_LANGUAGES } from '@algomart/schemas'
+import { SUPPORTED_LANGUAGES } from '@algomart/schemas'
+import { useRouter } from 'next/router'
+import { createContext, useContext, useEffect, useMemo, useState } from 'react'
+
+import { useAuth } from '@/contexts/auth-context'
+import { AuthService } from '@/services/auth-service'
+import { getCookie, setLanguageCookie } from '@/utils/cookies-web'
+
+interface ILanguageContext {
+  language: string
+  updateLanguage: (language: string) => Promise<boolean>
+}
+
+const LanguageContext = createContext<ILanguageContext>({
+  language: DEFAULT_LANG,
+  updateLanguage: () => Promise.resolve(true),
+})
+
+export const useLanguage = () => useContext(LanguageContext)
+
+function getSupportedLanguage(lang: string) {
+  // if your language is supported, return
+  // otherwise check if your language without country is supported
+  // otherwise default to english
+
+  if (SUPPORTED_LANGUAGES.includes(lang)) return lang
+  const [langPrefix] = lang.split('-')
+  return SUPPORTED_LANGUAGES.includes(langPrefix) ? langPrefix : DEFAULT_LANG
+}
+
+export const LanguageProvider = ({
+  children,
+}: {
+  children: React.ReactNode
```

Comment:
Just for consistency, could import `ReactNode` from the already-present `react` import.